docs/manage/configuration.md

```diff
@@ -0,0 +1,283 @@
+---
+title: "Robot Configuration"
+linkTitle: "Robot Configuration"
+weight: 25
+type: "docs"
+description: "Details about configuring robots with Viam."
+tags: ["manage", "components"]
+---
+
+To be able to use a robot, you must first configure it.

```

Comment:
I've tried an alternative flow for the config file and the config tab sections:

- it starts by explaining how a configuration is used and how it works
- it sees the app-based flow as the default and explains that
- I think the way to include that you can also store the configuration on the robot itself is as a note that points to a section at the end of the page (see below). So users use the app to create the configuration and if they cannot have the robot connect to the internet, store it directly on the robot. That's the best user experience I think. Slack me if you want to discuss that more. 

```
A robot's configuration tells the code running the robot what *resources*, [hardware *components*](#components) and [software *services*](#services), it has access to, as well as any relevant parameters for those resources.

To configure your robot, go to the [Viam app](https://app.viam.com) and [create a new robot](http://localhost:1313/installation/install/) and follow the steps on the **SETUP** page.

The setup steps copy your robots credentials to your robot and store them at <file>/etc/viam.json</file>.
The credentials look like this:

```json
{"cloud":{"app_address":"https://app.viam.com:123","id":"123456ab-a1b2-12a3-a123-1a2b345cdefg6",
"secret":"ab0cd0efghijkl0m12no3pqrstuvw0xyza4bcd567891abcd"}}
```

When you turn on your robot, `viam-server` starts up and checks its local <file>/etc/viam.json</file> for its configuration information.
The robot uses these credentials to get its full config from the Viam app.
To begin with, that configuration is nearly empty.

When you completed the setup steps and successfully connected to your robot, go to the **CONFIG** tab.

{{< alert title="Note" color="note" >}}
Your robot does not need to stay connected to the Viam app after it has obtained its configuration file.
The configuration is cached locally.
If you cannot connect your robot to the internet at all, you can also [store the configuration file](#Local Setup) on the robot itself.
{{< /alert >}}

## The CONFIG tab

The **CONFIG** tab is the place to configure everything about your robot.

![The CONFIG tab of the Viam app with a box highlighting the CONFIG tab and the Mode selector (where you can select Builder or Raw JSON).](../img/configuration/config-tab.png)

You can use the mode selector to switch between **Builder** and **Raw JSON**:

- **Builder** mode provides a graphical interface for configuring your robot resources.
- **Raw JSON** mode provides a text editing field where you can write and edit the config manually.

Regardless of the mode you choose, Viam stores the configuration file in [JSON (JavaScript Object Notation)](https://en.wikipedia.org/wiki/JSON).

If you add components in **Builder** mode and click **Save Config** at the bottom of the screen, you can switch to **Raw JSON** and see the corresponding JSON generated by the builder.

{{% expand "An example config file for a robot with a board component, motor component, and vision service configured" %}}

```json
{
  "components": [
    {
      "name": "local",
      "type": "board",
      "model": "pi",
      "attributes": {},
      "depends_on": []
    },
    {
      "name": "my-motor",
      "type": "motor",
      "model": "gpio",
      "attributes": {
        "pins": {
          "a": "13",
          "b": "15"
        },
        "board": "local",
        "max_rpm": 120
      },
      "depends_on": [],
    }
  ],
  "services": [
    {
      "name": "detector",
      "type": "vision",
      "attributes": {
        "register_models": [
          {
            "parameters": {
              "segment_size_px": 200,
              "hue_tolerance_pct": 0.05,
              "detect_color": "#19FFD9"
            },
            "type": "color_detector",
            "name": "green_detector"
          }
        ]
      }
    }
  ],
  "modules": []
}
```

{{% /expand %}}

Your robot's **CONFIG** page has sub-tabs to configure:

- [components](#components): Components are the hardware of your robot.
- [services](#services): Services are the software that runs on your robot.
- [modules](#modules): Modular resources are a way to add resource types or models that are not built into Viam.
- [remotes](#remotes): Remote are a way to connect two separate robots so one can access the resources of the other.
- [processes](#processes): Processes automatically run specified scripts when the robot boots.
- [fragments](#fragments): Fragments are a way of sharing and managing identical configuration files (or parts of config files) across multiple robots.
- [frames](#frame-system): Frames hold reference frame information for the relative position of components in space.
<!--[authorization keys](#authnetwork) -->
```